bigtable/beam/workload-generator/src/main/java/bigtable/WorkloadGenerator.java

```diff
@@ -88,19 +81,36 @@ private static void cancelJob(BigtableWorkloadOptions options, DataflowPipelineJ
     client.updateJob(jobId, job);
   }
 
-  public static class ReadFromTableFn extends AbstractCloudBigtableTableDoFn<Long, Void> {
+  public static class ReadFromTableFn extends DoFn<Long, Void> {
+
+    private final String projectId;
+    private final String instanceId;
+    private BigtableDataClient bigtableDataClient;
+
+    public ReadFromTableFn(String projectId, String instanceId) {
+      this.projectId = projectId;
+      this.instanceId = instanceId;
+    }
+
+    @Setup
+    public void setup() throws IOException {
+      this.bigtableDataClient = BigtableDataClient.create(this.projectId, this.instanceId);
+      System.out.println("Connected to client.");
+    }
 
-    public ReadFromTableFn(CloudBigtableConfiguration config) {
-      super(config);
-      System.out.println("Connected to table.");
+    @Teardown
+    public void teardown() {
+      this.bigtableDataClient.close();
     }
 
     @ProcessElement
-    public void processElement(PipelineOptions po) throws IOException {
+    public void processElement(PipelineOptions po) {
       BigtableWorkloadOptions options = po.as(BigtableWorkloadOptions.class);
-      Scan scan = new Scan();
-      Table table = getConnection().getTable(TableName.valueOf(options.getBigtableTableId()));
-      table.getScanner(scan);
+      Query query = Query.create(options.getBigtableTableId());
+      ServerStream<Row> rows = this.bigtableDataClient.readRows(query);
```

Comment:
Please make sure to consume the stream

Reply:
Updated what I had. Please let me know if that is okay or if I need to actually use the data in the stream